### PR TITLE
Fainting and auto switching

### LINKED ISSUE #76: Pokemon should faint when HP reaches 0.
We need a `Transaction`  that can tell the battle that a pokemon has fainted.

- [x] Triggers when pokemon's reaches 0
- [x] Checks to see if any pokemon in the party have not fainted, if all are fainted, battle is over
- [x] Sends out next pokemon

### PR DESCRIPTION
**Pull request by dyc3**
This PR mainly features a new `FaintTransaction` and a new `SendOutTransaction`. Unfortunately, this also requires a slight change to how transactions are processed, because transactions needed a way to trigger other transactions.

closes #76 

Would prefer #42 to be merged in first

### CODE REVIEW

File: battle.go [hunk 1]
```diff
@@ -125,24 +126,68 @@ func (b *Battle) SimulateRound() []Transaction {
...
 		}
-	}
 
-	// process transations
-	for _, transaction := range transactions {
-		switch t := transaction.(type) {
-		case DamageTransaction:
-			(*t.Target).CurrentHP -= t.Damage
+		// process transations for this turn
+		for len(queue) > 0 {
```
Comment (carpenterd777) [inline battle.go:142]: It might be worth creating a `TransactionQueue` in the future to improve readability of code written here

Comment (dyc3) [reply battle.go:142]: I mean it's not _that_ hard to read is it?

File: battle_test.go [hunk 2]
```diff
@@ -256,3 +258,73 @@ func TestTurnPriority(t *testing.T) {
...
+			// The Squirtle was unfazed.
+		},
+		{
+			turn: transactions[1],
+			want: "Squirtle used Pound on Charmander for 1282 damage.",
+			// Ash watched in horror as his Charmander was obliterated from the
+			// battlefield. "Critical hit!" echoed the automated announcer. The
+			// Squirtle snarled, now covered in the entrails of his previous
+			// opponent. "OH GOD, WHAT THE FUCK!?" sobbed Ash, "Is my friend
+			// really gone forever? Please tell me I'm dreaming, this can't be real!"
```
Comment (carpenterd777) [inline battle_test.go:313]: absolutely wonderful narrative here, award-winning

**Review by carpenterd777 — APPROVED** [review]
working code and a show, what more could I ask for

File: battle.go (hunk 1 above)
Comment (carpenterd777) [reply battle.go:142]: no its not, but I might not hurt as we continue to extend its functionality

**Pull request merged**